### PR TITLE
Removing SoundManager2 dependency

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

**Comment by bennycode** [comment]
@mythsunwind and @Yserz are thinking about writing tests which can read from the JS dev console if a sound started to play. So there needs to be a test which checks for this line:
`@logger.log @logger.levels.INFO, "Playing sound '#{audio_id}'", audio`

**Comment by gregor** [comment]
Please take another look. Unit tests have been added. Positive cases are x'ed out due to https://github.com/karma-runner/karma/issues/2320.

File: package.json [hunk 1]
```diff
@@ -16,25 +16,25 @@
...
     "karma-chrome-launcher": "1.0.1",
-    "karma-coverage": "1.0.0",
+    "karma-coverage": "1.1.1",
     "karma-jasmine": "1.0.2",
-    "load-grunt-tasks": "3.5.0",
-    "request": "2.72.0",
-    "sinon": "1.17.4"
+    "load-grunt-tasks": "3.5.2",
+    "request": "2.74.0",
+    "sinon": "1.17.5"
```
Comment (lipis) [inline package.json:37]: scary :D

File: app/script/audio/AudioRepository.coffee [hunk 2]
```diff
@@ -58,103 +59,154 @@ class z.audio.AudioRepository
...
-    @ringing_from_them = @_init_sound z.audio.AudioType.INCOMING_CALL, "#{AUDIO_PATH}/ringing_from_them.mp3"
-    @talk_later = @_init_sound z.audio.AudioType.TALK_LATER, "#{AUDIO_PATH}/talk_later.mp3"
-
-  # Use Amplify to subscribe to all audio playback related events.
-  _subscribe_to_audio_events: ->
-    amplify.subscribe z.event.WebApp.AUDIO.PLAY, @, @_play
-    amplify.subscribe z.event.WebApp.AUDIO.PLAY_IN_LOOP, @, @_play_in_loop
-    amplify.subscribe z.event.WebApp.AUDIO.STOP, @, @_stop
+  init: (skip_preloading = false) =>
+    @_init_sounds()
```
Comment (herrmannplatz) [inline app/script/audio/AudioRepository.coffee:66]: I would vote for `preload` with default value `false`

File: app/script/audio/AudioRepository.coffee [hunk 3]
```diff
@@ -58,103 +59,154 @@ class z.audio.AudioRepository
...
+    .then (audio_element) =>
+      @_play audio_id, audio_element, play_in_loop
+    .then (audio_element) =>
+      if play_in_loop
+        @logger.log @logger.levels.INFO, "Looping sound '#{audio_id}'", audio_element
+        @currently_looping[audio_id] = audio_id
+      else
+        @logger.log @logger.levels.INFO, "Playing sound '#{audio_id}'", audio_element
+    .catch (error) =>
+      switch error.type
```
Comment (herrmannplatz) [inline app/script/audio/AudioRepository.coffee:92]: do we really need this excessive logging?

File: app/script/audio/AudioRepository.coffee [hunk 4]
```diff
@@ -58,103 +59,154 @@ class z.audio.AudioRepository
...
-  @param is_first_time [Boolean] Is this the initial call or an on finish loop
   ###
-  _play_in_loop: (audio_id, is_first_time = true) ->
-    audio = soundManager.getSoundById audio_id
+  Initialize all sounds.
+  @private
+  ###
+  _init_sounds: ->
+    _create_sound = =>
+      audio_element = new Audio()
```
Comment (herrmannplatz) [inline app/script/audio/AudioRepository.coffee]: there is no need to define this inside the init function

**Comment by gregor** [comment]
@mythsunwind To simplify the logging the final message to check for is `Playing sound '{audio_id}' (in loop: '#{true|false}')`

**Pull request merged**